src/main/scala/io/gitlab/agluszak/tickets/controllers/ReservationController.scala

```diff
@@ -0,0 +1,72 @@
+package io.gitlab.agluszak.tickets.controllers
+
+import akka.http.scaladsl.marshalling.ToResponseMarshallable
+import akka.http.scaladsl.model.StatusCodes.{NotFound, OK, _}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import io.gitlab.agluszak.tickets.http.requests.ReservationCreateRequest
+import io.gitlab.agluszak.tickets.http.responses.{ApiError, ReservationCreateResponse}
+import io.gitlab.agluszak.tickets.services.ReservationService
+import io.gitlab.agluszak.tickets.{Protocol, ReservationIdTag}
+import pl.iterators.kebs.tagged._
+import pl.iterators.kebs.unmarshallers._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+
+
+class ReservationController(reservationService: ReservationService) extends Protocol {
+
+  val routes: Route =
+    pathPrefix("reservations") {
+      pathEndOrSingleSlash {
+        (post & entity(as[ReservationCreateRequest])) { (request) =>
+          complete {
+            reservationService.create(request).map[ToResponseMarshallable] {
+              case ReservationCreateResponse.AlreadyReserved(seats) =>
+                BadRequest ->
```

Comment:
I've been thinking about it for a while (https://softwareengineering.stackexchange.com/a/341824). 

I'm not very proud of throwing validation errors and business logic errors into the same bag.  The boundary I would set between one type of errors and the other is when the client could have known beforehand that the request was not going to work (that means that it could have been prevented by frontend). So 400 would be fine for requests which are *structurally* incorrect, for example not providing any seat numbers or providing diverging numbers of tickets and seats.

I think this also answers https://github.com/agluszak/touk-2019/pull/1#discussion_r327972478